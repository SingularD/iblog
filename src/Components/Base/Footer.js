import React from 'react'
import PropTypes from 'prop-types'

class Footer extends React.Component{
  render() {
    return (
      <div className="footer-container col-12 row justify-content-between">
        <div className="footer-page-turning col-12 row">
          <div className="col-4 text-left">
            <a href="#">上一页</a>
          </div>
          <div className="col-4 text-center">
            <a href="#">归档</a>
          </div>
          <div className="col-4 text-right">
            <a href="#">下一页</a>
          </div>
        </div>
      </div>
    )
  }
}
export default Footer
