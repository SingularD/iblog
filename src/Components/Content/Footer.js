import React from 'react'
import { Link } from "react-router-dom";

class Footer extends React.Component{
  render() {
    return (
      <div className="footer-container col-12 row justify-content-between">
        <div className="footer-page-turning col-12 row">
          <div className="col-4 text-left">
            <a href="#">上一页</a>
          </div>
          <div className="col-4 text-center">
            <Link to="/archive">归档</Link>
          </div>
          <div className="col-4 text-right">
            <a href="#">下一页</a>
          </div>
        </div>
      </div>
    )
  }
}
export default Footer
