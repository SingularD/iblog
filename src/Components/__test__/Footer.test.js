import React from 'react'
import { shallow } from "enzyme";
import Footer from '../Content/Footer'

describe("检测翻页组件", () => {
  let wrapper = shallow(<Footer />)
  it("匹配快照", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("检测内容是否正常渲染", () => {
    expect(wrapper.find('.text-left a').text()).toEqual('上一页')
    expect(wrapper.find('.text-center a').text()).toEqual('归档')
    expect(wrapper.find('.text-right a').text()).toEqual('下一页')

  })
})
