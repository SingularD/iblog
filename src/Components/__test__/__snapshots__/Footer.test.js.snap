// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`检测翻页组件 匹配快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="footer-page-turning col-12 row"
      >
        <div
          className="col-4 text-left"
        >
          <a
            href="#"
          >
            上一页
          </a>
        </div>
        <div
          className="col-4 text-center"
        >
          <a
            href="#"
          >
            归档
          </a>
        </div>
        <div
          className="col-4 text-right"
        >
          <a
            href="#"
          >
            下一页
          </a>
        </div>
      </div>,
      "className": "footer-container col-12 row justify-content-between",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-4 text-left"
          >
            <a
              href="#"
            >
              上一页
            </a>
          </div>,
          <div
            className="col-4 text-center"
          >
            <a
              href="#"
            >
              归档
            </a>
          </div>,
          <div
            className="col-4 text-right"
          >
            <a
              href="#"
            >
              下一页
            </a>
          </div>,
        ],
        "className": "footer-page-turning col-12 row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              上一页
            </a>,
            "className": "col-4 text-left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "上一页",
              "href": "#",
            },
            "ref": null,
            "rendered": "上一页",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              归档
            </a>,
            "className": "col-4 text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "归档",
              "href": "#",
            },
            "ref": null,
            "rendered": "归档",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              下一页
            </a>,
            "className": "col-4 text-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "下一页",
              "href": "#",
            },
            "ref": null,
            "rendered": "下一页",
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="footer-page-turning col-12 row"
        >
          <div
            className="col-4 text-left"
          >
            <a
              href="#"
            >
              上一页
            </a>
          </div>
          <div
            className="col-4 text-center"
          >
            <a
              href="#"
            >
              归档
            </a>
          </div>
          <div
            className="col-4 text-right"
          >
            <a
              href="#"
            >
              下一页
            </a>
          </div>
        </div>,
        "className": "footer-container col-12 row justify-content-between",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-4 text-left"
            >
              <a
                href="#"
              >
                上一页
              </a>
            </div>,
            <div
              className="col-4 text-center"
            >
              <a
                href="#"
              >
                归档
              </a>
            </div>,
            <div
              className="col-4 text-right"
            >
              <a
                href="#"
              >
                下一页
              </a>
            </div>,
          ],
          "className": "footer-page-turning col-12 row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                上一页
              </a>,
              "className": "col-4 text-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "上一页",
                "href": "#",
              },
              "ref": null,
              "rendered": "上一页",
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                归档
              </a>,
              "className": "col-4 text-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "归档",
                "href": "#",
              },
              "ref": null,
              "rendered": "归档",
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                下一页
              </a>,
              "className": "col-4 text-right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "下一页",
                "href": "#",
              },
              "ref": null,
              "rendered": "下一页",
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
